### PR TITLE
added login with SSO

### PR DESCRIPTION
**Pull request by mekilis**

### CODE REVIEW

File: api/handlers/auth.go [hunk 1]
```diff
@@ -15,6 +15,71 @@ import (
 	"github.com/go-chi/render"
 )
 
+func (h *Handler) InitLoginSSO(w http.ResponseWriter, r *http.Request) {
+
+	configuration, err := config.Get()
+	if err != nil {
+		_ = render.Render(w, r, util.NewErrorResponse(err.Error(), http.StatusBadRequest))
+		return
+	}
```
Comment (jirevwe) [inline api/handlers/auth.go:24]: The handler already has a copy of the configuration. It should be h.A.Cfg

Comment (mekilis) [reply api/handlers/auth.go:24]: Noted, thank you.

File: services/login_sso.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	if !tokenResponse.Status {
+		log.Errorf("failed to redeem token in response: %+v", tokenResponse)
+		return nil, errors.New("failed to redeem token")
+	}
+
+	if tokenResponse.Data.Payload.Email == "" {
+		return nil, errors.New("no email in token response")
+	}
+	if tokenResponse.Data.Payload.OrganizationExternalID == "" {
+		return nil, errors.New("no email in token response")
```
Comment (danvixent) [inline services/login_sso.go:144]: I'm not sure if this error message is on purpose?

Comment (mekilis) [reply services/login_sso.go:144]: It's not. An oversight that will be corrected

**Review by jirevwe — APPROVED** [review]

**Pull request merged**